diff --git a/CMakeLists.txt b/CMakeLists.txt
index e5386fe..7728a6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,10 @@ project(openFPGALoader VERSION 0.1 LANGUAGES CXX)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_CXX_FLAGS_DEBUG "-g -Wall -Wextra")
+set(CMAKE_EXE_LINKER_FLAGS "-static-libstdc++ -static")
+set(BUILD_SHARED_LIBS OFF)
+
+# set_target_properties(icarus PROPERTIES LINK_SEARCH_END_STATIC 1)
 
 include(GNUInstallDirs)
 # By default: DATA_DIR="/usr/local/share"
@@ -15,7 +19,7 @@ add_definitions(-DDATA_DIR=\"${CMAKE_INSTALL_FULL_DATAROOTDIR}\")
 
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(LIBFTDI REQUIRED libftdi1)
-pkg_check_modules(LIBFTDIPP REQUIRED libftdipp1)
+# pkg_check_modules(LIBFTDIPP REQUIRED libftdipp1)
 pkg_check_modules(LIBUSB REQUIRED libusb-1.0)
 pkg_check_modules(LIBUDEV REQUIRED libudev)
 
@@ -88,7 +92,9 @@ target_link_libraries(openFPGALoader
 	${LIBUSB_LIBRARIES}
 	${LIBFTDI_LIBRARIES}
 	${LIBFTDIPP_LIBRARIES}
+	/usr/lib/libargp.a
 )
+set_target_properties(openFPGALoader PROPERTIES LINK_SEARCH_END_STATIC 1)
 
 # libftdi < 1.4 as no usb_addr
 if (${LIBFTDI_VERSION} VERSION_LESS 1.4)
